src/config.rs

```diff
@@ -62,10 +63,15 @@ pub struct Config {
 
 impl Config {
     fn new(config_path: String) -> Result<Self, ConfigError> {
-        let mut settings = config::Config::default();
-        settings
-            .merge(config::File::with_name(&config_path))
-            .unwrap();
-        settings.try_into()
+        config::Config::builder()
+            .set_default("db.url", "sqlite://db.sqlite?mode=rwc".to_string())?
+            .set_default("cors.allowed_methods", vec!["GET", "POST"])?
+            .set_default("cors.allowed_headers", vec!["ACCEPT", "CONTENT_TYPE"])?
+            .set_default("limits.global_requests_per_day", 10000)?
+            .set_default("link.max_length", 4096)?
+            .set_default("webapp.redirect_from_path", "/".to_string())?
+            .add_source(File::new(&config_path, FileFormat::Toml))
+            .build()?
+            .try_deserialize()
```

Comment:
Isn't it possible to keep the `Default` trait implementation and build on top of a default instance?

Reply:
Tried that first but it doesn't work.

Reply:
> In general looks good to me. Unrelated to the code, I would strongly prefer if you...
Good suggestion. Actually, I myself noticed that it is not a good practice as I also struggled to understand what that commit did without seeing the commit diff. Thanks for suggestion.

Reply:
Squashing is another option, which for this case can make sense. I usually prefer not to squash commits.

I'm fine to squash this time, but just in case, do you know the command `git rebase -i` (or `--interactive`)? It is very easy to change commit messages from previous commit (use the option `reword` to just do that). Here is a online tutorial that explains a bit of it, in case you are not familiar: https://hackernoon.com/beginners-guide-to-interactive-rebasing-346a3f9c3a6d

Reply:
Just let me know if you want to give it a try or if I should just squash-merge.

Reply:
Actually, I prefer to put one feature in one commit and that is why I wanted to squash commits but anyways I will follow your approach. 

I would go for squash-merge this time because.